src/components/inputs/dropdown-input.tsx

```diff
@@ -0,0 +1,88 @@
+import { useTheme } from 'native-base';
+import React, { useState } from 'react';
+import { FlatList, Text, TouchableOpacity, View } from 'react-native';
+import Icon from 'react-native-vector-icons/FontAwesome';
+
+import { DropdownInputProps, DropdownOptionProps } from '../../types';
+
+import { useDropdownInputStyles } from './input.styles';
+
+const DropdownInput: React.FC<DropdownInputProps> & {
+  Option: React.FC<DropdownOptionProps>;
+} = ({ children, disabled = false, label, onValueChange, selectedValue }) => {
+  const [isDropdownVisible, setDropdownVisible] = useState(false);
+  const dropdownStyles = useDropdownInputStyles();
+  const theme = useTheme();
+
+  const toggleDropdown = () => {
+    if (!disabled) {
```

Comment:
Why is this needed here? I would assume this method should not even be possible to invoke when control is disabled.